google-cloud-spanner/src/main/java/com/google/cloud/spanner/StructReader.java

```diff
@@ -123,6 +123,18 @@ public interface StructReader {
    */
   long getLong(String columnName);
 
+  /**
+   * @param columnIndex index of the column
+   * @return the value of a non-{@code NULL} column with type {@link Type#float32()}.
+   */
+  float getFloat(int columnIndex);
```

Comment:
New methods in an interface should have a default implemention that throws an `UnsupportedOperationException`. See the example for `getJson` in this file.

The reason is that:
1. The current implementation will cause a compilation error for any external user that implements this interface, for example in tests. That is a breaking change.
2. Adding a default implementation defers any error until runtime and the actual usage of the method, which makes this change non-breaking.

Reply:
Thanks for the context. Makes sense, made the change.

Reply:
Can you PTAL at the CLIRR differences too to verify this? I had to add the Value.float32 to the ignored differences, but this seems to be the first addition in this class.

Reply:
The clirr diffs look fine. This is however not the first addition to this class. See for example the addition of `jsonb`.